packages/edit-site/src/components/add-new-template/add-custom-template-modal.js

```diff
@@ -145,14 +145,25 @@ function SuggestionList( { entityForSuggestions, onSelect } ) {
 		debouncedSearch
 	);
 	const { labels } = entityForSuggestions;
+	// If there are less suggestions than our initial ones with no `search`(10),
+	// there is no need to show the `SearchControl`. We use a `ref` to track
+	// this because the initial search request is enough to let us decide.
```

Comment:
```suggestion
	// this because the initial search request isn't enough to let us decide.
```

Assuming this was an intended comment, can you give us more details on why checking `suggestions?.length > 9` isn't enough?

As you know, the setting `ref` doesn't trigger re-render, so we might end up with a stale component state.

P.S. React docs also don't recommend writing or reading ref during the render. See the "Pitfall" section - https://beta.reactjs.org/apis/useref#referencing-a-value-with-a-ref.

Reply:
The comment is right because when we first render the component we know for sure that it has available items ([checks are in parent component](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/add-new-template/utils.js#L289)) and the `search` term is `empty`. When we don't have a `search term` the component renders some initial suggestions, which are `ten` in our case.

So if the initial request on first render(our 10 suggestions) fetches less than that, we know that there are no more and it's okay not to show the search control.  Without using a `ref` for this, I couldn't find a way to make this handle properly the cases where we clear the search but there are still some suggestions(debounced search and we set the suggestions when the request has resolved). Couldn't find a good way to differentiate these and we need a value set only once. Makes sense?

Do you think we could do it differently? Also I guess the comment should be updated if we go with this approach?

Reply:
Maybe use state instead? Something like this works as well in my tests.

```js
const [ showSearchControl, setShowSearchControl ] = useState( false );
useEffect( () => {
		if ( ! showSearchControl && suggestions?.length > 9 ) {
			setShowSearchControl( true );
		}
}, [ showSearchControl, suggestions ] );
```

Reply:
This reminds me of [this recent comment](https://github.com/WordPress/gutenberg/pull/42116#discussion_r912946663).

Is the following an apt summary of the problem?

_We'd like to only show the search bar if there are enough suggestions to justify it (10+), but this should only matter upon mounting — after that, if the user narrows down the list of suggestions by searching, we should never hide the search bar._

If so, what about a simpler inline state update:

```js
const [ hideSearchControl, setHideSearchControl ] = useState( false );
if ( ! hideSearchControl && suggestions?.length > 9 ) {
    setHideSearchControl( true );
}
return (
    <> ...
```

Reply:
> We'd like to only show the search bar if there are enough suggestions to justify it (10+), but this should only matter upon mounting....

That's exactly the case, thanks for sharing. Thank you both for the suggestions - I updated the code.